### PR TITLE
Improve error if name of an exe is not in modules

### LINKED ISSUE #5787: Confusing message about non-existent module
With this in my `dune` file

```
(executable
  (name dlist)
  (libraries base)
  (modules doubly_linked))
```

I get the following error message

```
File "dune", line 20, characters 8-13:
20 |   (name dlist)
             ^^^^^
Error: Module "Dlist" doesn't exist.
```

## Desired behavior

A clearer error message that guides the user towards the solution — fixing the `modules` list.

```diff
  File "dune", line 20, characters 8-13:
  20 |   (name dlist)
               ^^^^^
- Error: Module "Dlist" doesn't exist.
+ Error: Module "Dlist" exists but is not part of the modules list.
```

### PR DESCRIPTION
**Pull request by emillon**
Closes #5787

### CODE REVIEW

**Comment by rgrinberg** [comment]
What's the error when the module is indeed listed but the source file doesn't exist?

**Comment by emillon** [comment]
In that case we get "module A does not exist", which I think is the right one. I added a test for this.

File: CHANGES.md [hunk 1]
```diff
@@ -67,6 +67,9 @@
 - The test suite for Coq now requires Coq >= 8.16 due to changes in the
   plugin loading mechanism upstream (which now uses findlib).
 
+- Improve error message when the (name) of an executable is not part of the
+  (modules) listed for it. (#6169, fixes #5787, @emillon)
```
Comment (rgrinberg) [inline CHANGES.md:71]: FYI we don't usually document changes to error messages. Although of course I won't object if you do it.

**Review by rgrinberg — APPROVED** [review]

File: CHANGES.md (hunk 1 above)
Comment (emillon) [reply CHANGES.md:71]: Right, I'm reverting that part.

**Pull request merged**